### PR TITLE
[e2e] Using ganache requests to getBalance and getAccounts

### PR DESCRIPTION
**Pull request by seaona**
## Explanation
**Context**: sometimes we need to validate tests against the current ETH balance or the account addresst. We used to hardcode this values.

**What**: with this PR we introduce the Ganache methods for getting the accounts and getting the balance of the account 0, so we do not need to hardcode balances/accounts anymore. Benefits:
- If we add more transactions in the previous steps, we won't have to update the balance manually
- We are indirectly adding more test coverage, as we are making sure that what we receive from the node is correctly displayed on the UI
- This work can unlock potential test scenarios using other Ganache methods like `setAccountNonce`etc. 

## Screenshots/Screencaps

n/a

### Before
We look for this specific balance value on the UI:
`text: '24.9977 ETH',`

### After
We get the current balance value and look for that value on the UI:
`const balance = await ganacheServer.getBalance();`
`text: ${balance} ETH,`

## Manual Testing Steps
Run the affected tests manually on your local:
- `yarn test:e2e:single test/e2e/tests/send-hex-address.spec.js --browser=chrome --leave-running=true`
- `yarn test:e2e:single test/e2e/tests/metamask-responsive-ui.spec.js --browser=chrome --leave-running=true`

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/ganache.js [hunk 1]
```diff
@@ -23,6 +23,27 @@ class Ganache {
...
+  async getBalance() {
+    const accounts = await this.getAccounts();
+    const balanceHex = await this.getProvider().request({
+      method: 'eth_getBalance',
+      params: [accounts[0], 'latest'],
+    });
+    const balanceInt = parseInt(balanceHex, 16) / 10 ** 18;
+
+    const balanceFormatted =
+      balanceInt % 1 === 0 ? balanceInt : balanceInt.toFixed(4);
```
Comment (seaona) [inline test/e2e/ganache.js:42]: If the balance is a float number, we round it up to 4 decimals to match UI.
If it's an integer, do nothing

File: test/e2e/helpers.js [hunk 2]
```diff
@@ -145,6 +145,7 @@ async function withFixtures(options, testSuite) {
       driver: driverProxy ?? driver,
       mockServer,
       contractRegistry,
+      ganacheServer,
```
Comment (seaona) [inline test/e2e/helpers.js:148]: we pass ganacheServer to the testsuite to be able to access its methods

File: test/e2e/tests/metamask-responsive-ui.spec.js [hunk 3]
```diff
@@ -104,9 +104,10 @@ describe('MetaMask Responsive UI', function () {
         await driver.press('#confirm-password', driver.Key.ENTER);
 
         // balance renders
+        const balance = await ganacheServer.getBalance();
```
Comment (seaona) [inline test/e2e/tests/metamask-responsive-ui.spec.js:107]: here we access the current balance of the acocunt, instead of hardcoding the balance

**Review by jpuri — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM! Great idea

**Pull request merged**